### PR TITLE
Estcard support improved

### PR DESCRIPTION
**Pull request by lenar**
Fixes small kinks in Estcard support

### CODE REVIEW

File: src/Protocol/ECommerce.php [hunk 1]
```diff
@@ -252,8 +264,8 @@ protected function generateSignature(array $data, $encoding = 'UTF-8')
             );
 
             $data['receipt_no'] = ProtocolHelper::mbStrPad($data['receipt_no'], 6, "0", STR_PAD_LEFT, $encoding);
+            $data['msgdata'] = ProtocolHelper::mbStrPad($data['msgdata'], 40, " ", STR_PAD_RIGHT, $encoding, $data['respcode'] === self::PAYMENT_RESPONSE_ABORT);
```
Comment (renekorss) [inline src/Protocol/ECommerce.php]: We could do:

````php
$data['msgdata'] = ProtocolHelper::mbStrPad(
    $data['respcode'] === self::PAYMENT_RESPONSE_ABORT ? " " : $data['msgdata'], 
    40, 
    " ", 
    STR_PAD_RIGHT, 
    $encoding
);
````
and leave `ProtocolHelper::mbStrPad` as it is. Just wondering if we can avoid `ProtocolHelper::mbStrPad` single responsibility principle violation and not make it more complicated?

**Comment by renekorss** [comment]
Also, could you fix Estcard URL in tests too? 
https://github.com/renekorss/Banklink/blob/master/tests/EstcardTest.php#L20

And remove `additionalinfo` from
https://github.com/renekorss/Banklink/blob/master/tests/EstcardTest.php#L56
since it will fail tests, because request dosen't have it anymore. 
It's weird that it has to be in mac calculation. Did you tested it on Estcard test system? I used pangalink.net application and had no problems.

File: src/Protocol/ECommerce.php (hunk 1 above)
Comment (lenar) [inline src/Protocol/ECommerce.php]: Could be solved this way too. Although it feels more hacky to me.
Only I would add one additional check:

```
$data['msgdata'] = ProtocolHelper::mbStrPad(
    $data['respcode'] === self::PAYMENT_RESPONSE_ABORT && strlen($data['msgdata']) == 0 ? ' ' : $data['msgdata'],
    40,
    " ",
    STR_PAD_RIGHT,
    $encoding
);
```

**Comment by lenar** [comment]
> It's weird that it has to be in mac calculation. Did you tested it on Estcard test system? I used pangalink.net application and had no problems.

Yes, tested against WebPOS official test environment and actual comment from @estcard.ee regarding the issue confirmed that if additionalinfo is included in request it is also included in MAC calculation. Supposedly this field is used mostly by airline companies to pass a 'reference number' which is mandatory for them. The comment also agreed that the specifications are somewhat incomplete and need improving.

**Comment by renekorss** [comment]
Okay. Very good. Can you make changes I wrote? Then I would merge and publish new ver.

File: src/Protocol/ECommerce.php (hunk 1 above)
Comment (renekorss) [inline src/Protocol/ECommerce.php]: Yes, it feels hacky. But mbStrPad will be better. Will refactor this later. Maybe some more suggestions? :)

**Comment by renekorss** [comment]
Thank you.

**Comment by renekorss** [comment]
Released in [v2.0.1](https://github.com/renekorss/Banklink/releases/tag/v2.0.1).

**Pull request merged**